### ISSUE TITLE
adminUsername Should Not Be A Literal conflict with Bicep in nested deployment

### ISSUE THREAD

**Issue by lr90** [Azure/arm-ttk#349]
Odd one this when trying to pass down the adminUsername through modules.

main.bicep

```
targetScope = 'resourceGroup'

param adminUsername string
param location string

module vm 'vm.bicep' ={
  name: 'vmDeploy'
  params:{
    adminUsername: adminUsername
    location: location
  }
}
```
and vm.bicep
```
param adminUsername string
param location string
param now string = utcNow()
param nicId string = 'stub to simplify repo'
param vmName string = 'myvm'
param vmSize string = 'Standard_B2ms'

var adminPassword = '${uniqueString(resourceGroup().id, now)}-X'

resource vm 'Microsoft.Compute/virtualMachines@2020-06-01' = {
  name: vmName
  location: location
  properties: {
    hardwareProfile: {
      vmSize: vmSize
    }
    networkProfile: {
      networkInterfaces: [
        {
          id: nicId
        }
      ]
    }
    osProfile: {
      computerName: vmName
      adminUsername: adminUsername
      adminPassword: adminPassword
      windowsConfiguration: {
        provisionVMAgent: true
        enableAutomaticUpdates: true
        patchSettings: {
          patchMode: 'AutomaticByOS'
        }
      }
    }
    storageProfile: {
      imageReference: {
        publisher: 'MicrosoftWindowsServer'
        offer: 'WindowsServer'
        sku: '2019-Datacenter'
        version: 'latest'
      }
      osDisk: {
        name: '${vmName}-os'
        createOption: 'FromImage'
        managedDisk: {
          storageAccountType: 'StandardSSD_LRS'
        }
      }
    }
  }
}
```
transpiling and testing vm.bicep is fine but transpiling main.bicep and testing gives the 'adminUsername Should Not Be A Literal' error.

Looking at the resulting json file this seems to be due to it's appearance in the parameters of the nested deployment:
```
  "resources": [
    {
      "type": "Microsoft.Resources/deployments",
      "apiVersion": "2019-10-01",
      "name": "vmDeploy",
      "properties": {
        "expressionEvaluationOptions": {
          "scope": "inner"
        },
        "mode": "Incremental",
        "parameters": {
          "adminUsername": {
            "value": "[parameters('adminUsername')]"
          },
          "location": {
            "value": "[parameters('location')]"
          }
```

**Comment by bmoore-msft** [comment]
I think this is related to our handling of nested deployments though this may be a slightly different twist on the params passed in (similar to #318 )

**Comment by lr90** [comment]
I suspect I am seeing the same issue with 'DeploymentTemplate Must Not Contain Hardcoded Uri' which is appearing when the deployment is nested. Should I raise another issue with minimal repo code?

**Comment by lr90** [comment]
Scrub that the URI test was my bad but inworking through a repro I've found something rather different and raised #351

**Comment by bmoore-msft** [comment]
fixed in #356
